src/main/java/adris/altoclef/tasks/GetResourceTask.java

```diff
@@ -0,0 +1,87 @@
+package adris.altoclef.tasks;
```

Comment:
This class is a good idea, but that's not exactly how this system handles situations like this.

The functionality of this class should be split into two parts:

1) A parameter for "isStarvingHungerThreshold" and "foodToCollectWhenStarving" (names may vary) in [Settings.java](https://github.com/adrisj7-AltoClef/altoclef/blob/main/src/main/java/adris/altoclef/Settings.java). By default this should be set to 0 or -1 or something to disable it
2) In [FoodChain.java](https://github.com/adrisj7-AltoClef/altoclef/blob/main/src/main/java/adris/altoclef/tasksystem/chains/FoodChain.java), at the very bottom of `getPriority`, check if we have less food units than the value in settings. If so, return a priority of something like 55 (will need to fix these magic numbers) and run `setTask(new CollectFoodTask)` with the parameter from settings.

After that, GetResourceTask shouldn't be needed anymore.

I would also consider when it's appropriate to need to get food. Is it when the player is starving/hungry, or when the player doesn't have enough food in its inventory?

Reply:
i'm actually still figuring out how the system works, so yes i'll do it this way.

Reply:
> A parameter for "isStarvingHungerThreshold" and "foodToCollectWhenStarving" (names may vary) in Settings.java.

I was already thinking to add the parameters in the settings but I didn't do it for some reason.

> In FoodChain.java, at the...

Ok, now I will do as you say. Before I didn't quite understand how the system worked so I did it this way and it worked, now I will change it.